packages/core/utils/lib/sanitize/visitors/remove-restricted-relations.js

```diff
@@ -42,12 +43,18 @@ module.exports = auth => async ({ data, key, attribute }, { remove, set }) => {
   };
 
   const isMorphRelation = attribute.relation.toLowerCase().startsWith('morph');
+  const isCreatorRelation = [CREATED_BY_ATTRIBUTE, UPDATED_BY_ATTRIBUTE].includes(key);
 
   // Polymorphic relations
   if (isMorphRelation) {
     await handleMorphRelation();
   }
 
+  // Creator relations
+  else if (isCreatorRelation && schema.options.populateCreatorFields) {
```

Comment:
I'm not sure we should make this check on `schema.options.populateCreatorFields` here.

When enabled, this option modifies the `private` state of the creators. Since private fields are already removed from the data in a dedicated sanitation step (`removePrivate` visitor), I don't think we should couple the creator logic here.

Maybe for now, we could make a very specific exception for those creator fields relations while we think about introducing another layer here (basically just removing this && schema.options.populateCreatorFields).

Finally, I think you can return early instead of doing nothing here. 

What do you think?

Reply:
I had it that way originally, but added the check after talking with @alexandrebodin . While private fields will be removed in a later sanitization step, it feels like if it's not enabled those fields should be treated with handleRegularRelation (which will then remove it based on if the user has the true permission or not). That is, not necessarily trusting the following step to do its job properly, and removing it here if we're not certain.

Regarding returning early, I agree, I didn't want to change the if-else flow that was already there. I will update handleMorphRelation and handleRegularRelation as well. Do we have a policy or preference on switch statements? :)

Reply:
Thanks for the detailed explanation :+1: